sys/pool_alloc.c

```diff
@@ -0,0 +1,235 @@
+#include <stdc.h>
+#include <vm.h>
+#include <physmem.h>
+#include <queue.h>
+#include <bitstring.h>
+#include <common.h>
+#include <pool_alloc.h>
+
+#define PI_MAGIC_WORD 0xcafebabe
+
+#define PP_FREED_WORD 0xdeadbeef
+
+#define GET_PI_AT_IDX(slab, i, size)                                           \
+  (pool_item_t *)((unsigned long)slab + (slab->ph_start) +                     \
+                  (i) * (size + sizeof(pool_item_t)))
+
+#define ALIGN_TO_BINARY_POW(x, a) ((((x) >> (a)) + 1) << (a))
+
+#define PALLOC_DEBUG 0
+
+#if defined(PALLOC_DEBUG) && PALLOC_DEBUG > 0
+#define PALLOC_DEBUG_PRINT(text, args...)                                      \
+  kprintf("PALLOC_DEBUG: %s:%d:%s(): " text, __FILE__, __LINE__, __func__,     \
+          ##args)
+#else
+#define PALLOC_DEBUG_PRINT(text, args...)
+#endif
+
+typedef struct pool_item {
+  unsigned long pi_guard_number; /* PI_MAGIC_WORD by default, normally isn't */
+  /* changed, so if it has another value, memory */
+  /* corruption has taken place */
+  pool_slab_t *pi_slab; /* pointer to slab containing this item */
+  unsigned long pi_data[0];
+} pool_item_t;
+
+pool_slab_t *create_slab(size_t size, pool_t *pool,
```

Comment:
`create_slab`, `destroy_slab` and `slab_alloc` are not a part of the interface, right? Make sure to mark these functions as `static`, so that they are inaccessible from other files.